rust/linux-client/src/main.rs

```diff
@@ -102,6 +102,37 @@ fn get_system_default_resolvers_resolv_conf() -> Result<Vec<IpAddr>> {
     Ok(nameservers)
 }
 
+fn get_system_default_resolvers_resolvectl() -> Result<Vec<IpAddr>> {
+    // Unfortunately systemd-resolved does not have a machine-readable
+    // text output for this command: <https://github.com/systemd/systemd/issues/29755>
+    //
+    // The officially supported way is probably to use D-Bus.
+    let output = std::process::Command::new("resolvectl")
+        .arg("dns")
+        .output()
+        .context("Failed to run `resolvectl dns` and read output")?;
+    if !output.status.success() {
+        anyhow::bail!("`resolvectl dns` returned non-zero exit code");
+    }
+    let output = String::from_utf8(output.stdout).context("`resolvectl` output was not UTF-8")?;
+    Ok(parse_resolvectl_output(&output))
+}
+
+/// Parses the text output of `resolvectl dns`
+///
+/// Cannot fail. If the parsing code is wrong, the IP address vec will just be incomplete.
+fn parse_resolvectl_output(s: &str) -> Vec<IpAddr> {
+    let mut v = vec![];
+    for line in s.lines() {
+        for word in line.split(' ') {
+            if let Ok(addr) = IpAddr::from_str(word) {
+                v.push(addr);
+            }
+        }
+    }
+    v
```

Comment:
```suggestion
    s.lines().flat_map(|l| line.split(' ')).filter_map(|w| IpAddr::from_str(w).ok()).collect()
```

I think this should do the trick too :)

Reply:
I have no experience with Ansible either, I'm using it for lack of a local CI runner, so I can reset a VM to a "before Firezone was installed" state. All it does is copy a few files and set capabilities on the exe in a reproducible way.